firewallns.py

```diff
@@ -0,0 +1,800 @@
+# stdlib
+import json
+import ipaddress
+from pathlib import Path
+from collections import deque
+from typing import Deque, List, Tuple
+# lib
+from cloudcix.rcc import comms_ssh, CouldNotConnectException
+# local
+from .controllers import FirewallNamespace
+from utils import JINJA_ENV, check_template_data
+
+__all__ = [
+    'build',
+]
+
+SUCCESS_CODE = 0
+
+
+def complete_rule(rule, iiface, oiface, namespace, table):
+    v = '' if rule['version'] == '4' else '6'
+
+    # input interface line
+    iif = f'iifname {iiface}' if iiface not in [None, 'any'] else ''
+
+    # output interface line
+    oif = f'oifname {oiface}' if oiface not in [None, 'any'] else ''
+
+    # sort the `destination` rule format
+    if rule['destination'] is None or 'any' in rule['destination']:
+        daddr = ''
+    else:
+        daddr = f'ip{v} daddr ' + '{' + ','.join(rule['destination']) + '}'
+
+    # sort the `source` rule format
+    if rule['source'] is None or 'any' in rule['source']:
+        saddr = ''
+    else:
+        saddr = f'ip{v} saddr ' + '{' + ','.join(rule['source']) + '}'
+
+    # sort the `port` rule format
+    if rule['port'] is None or rule['protocol'] == 'any':
+        dport = ''
+    else:
+        dport = 'dport ' + '{' + ','.join(rule['port']) + '}'
+
+    # rule protocol and port statement, also gather protocols that require to define chains in config
+    application = None
+    if rule['protocol'] == 'any':
+        proto_port = f'{rule["action"]}'
+    elif rule['protocol'] == 'icmp':
+        proto_port = f'jump icmp{v}_{rule["action"]}'
+        application = f'icmp{v}_{rule["action"]}'
+    elif rule['protocol'] == 'dns':
+        proto_port = f'jump dns_{rule["action"]}'
+        application = f'dns_{rule["action"]}'
+    elif rule['protocol'] == 'vpn':
+        proto_port = f'jump vpn_{rule["action"]}'
+        application = f'vpn_{rule["action"]}'
+    else:
+        proto_port = f'{rule["protocol"]} {dport} {rule["action"]}'
+
+    # log
+    log = f'log prefix "{namespace} Table: {table}" level debug group 0' if rule['log'] is True else ''
+
+    return f'{iif} {oif} {saddr} {daddr} {log} {proto_port}', application
+
+
+def dnat_rule(nat):
+    rule_line = f'iifname "{nat["iiface"]}" ' if nat['iiface'] is not None else ''
+    rule_line = f'{rule_line}ip daddr {nat["public"]} dnat to {nat["private"]} '
+    return rule_line
+
+
+def snat_rule(nat):
+    rule_line = f'oifname "{nat["oiface"]}" ' if nat['oiface'] is not None else ''
+    rule_line = f'{rule_line}ip saddr {nat["private"]} snat to {nat["public"]} '
+    return rule_line
+
+
+def build(
+        namespace: str,
+        table: str,
+        priority: int,
+        config_file=None,
+        rules=None,
+        nats=None,
+) -> Tuple[bool, str]:
+    """
+    description: |
+        1. Creates a /tmp/firewallns_<namespace>_<table>.conf file with nftable table config
+        2. Validates the nft file `sudo nft -c -f /tmp/firewallns_<namespace>_<table>.conf`
```

Comment:
`sudo` should not be required in any payload request. The user sending the request via rcc.comms_ssh() should have the permissions to execute the payload.